### PR TITLE
port to slim v4

### PR DESCRIPTION
**Pull request by petrelharp**
With these changes, tests now pass using the new version of SLiM on my computer at least. Mostly this just required changing "generation" to "tick"; there was only one thing that was hard to track down.

### CODE REVIEW

**Comment by petrelharp** [comment]
AFAICT, codecov is just being confused here. The only lines it says are not being covered now are lines like "don't test this on windows".

**Comment by andrewkern** [comment]
okay, well to review this properly i'm going to have to start by reading the new [time units](https://tskit.dev/pyslim/docs/latest/time_units.html) section!

File: stdpopsim/slim_engine.py [hunk 1]
```diff
@@ -503,20 +503,22 @@
...
+                " ("+pop_names[pop]+") at tick "+g+", but only "+
                 N_g+" individuals will be alive.");
         }
 
-        sim.registerLateEvent(NULL,
-            "{dbg(self.source); " +
-            "inds=p"+pop+".sampleIndividuals("+n+"); " +
-            "sim.treeSeqRememberIndividuals(inds);}",
-            g, g);
+        if (n > 0) {
```
Comment (andrewkern) [inline stdpopsim/slim_engine.py:510]: what happens if n <= 0?

**Review by andrewkern — APPROVED** [review]
looks like a simple enough conversion to v4, but there is definitely going to be breakage. 

should average generation time be recorded in the metadata (as in the pyslim examples)? currently everything is WF so perhaps not?

**Comment by petrelharp** [comment]
> looks like a simple enough conversion to v4, but there is definitely going to be breakage.

Where?

> should average generation time be recorded in the metadata (as in the pyslim examples)? currently everything is WF so perhaps not?

No, everything is WF, so one tick = one generation.  (well, actually it equals `Q` generations...) It's something we'll have to deal with if/when we provide nonWF recipes, but that's a hurdle for the future.

File: stdpopsim/slim_engine.py (hunk 1 above)
Comment (petrelharp) [reply stdpopsim/slim_engine.py:510]: https://github.com/MesserLab/SLiM/issues/345 happens

**Comment by andrewkern** [comment]
okay sounds good @petrelharp!

**Pull request merged**